api_tests/files/views/test_file_detail.py

```diff
@@ -478,104 +496,114 @@ def test_listing(self):
+            auth=user.auth,
         )
-        assert_equal(res.status_code, 200)
-        assert_equal(res.json['data']['id'], '1')
+        assert res.status_code == 200
+        assert res.json['data']['id'] == '1'
 
-    def test_read_only(self):
-        assert_equal(self.app.put(
-            '/{}files/{}/versions/1/'.format(API_BASE, self.file._id),
+        # test_read_only
+        assert app.put(
+            '/{}files/{}/versions/1/'.format(API_BASE, file._id),
             expect_errors=True,
-            auth=self.user.auth,
-        ).status_code, 405)
+            auth=user.auth,
+        ).status_code == 405
 
-        assert_equal(self.app.post(
-            '/{}files/{}/versions/1/'.format(API_BASE, self.file._id),
+        assert app.post(
+            '/{}files/{}/versions/1/'.format(API_BASE, file._id),
             expect_errors=True,
-            auth=self.user.auth,
-        ).status_code, 405)
+            auth=user.auth,
+        ).status_code == 405
 
-        assert_equal(self.app.delete(
-            '/{}files/{}/versions/1/'.format(API_BASE, self.file._id),
+        assert app.delete(
+            '/{}files/{}/versions/1/'.format(API_BASE, file._id),
             expect_errors=True,
-            auth=self.user.auth,
-        ).status_code, 405)
-
-
-class TestFileTagging(ApiTestCase):
-    def setUp(self):
-        super(TestFileTagging, self).setUp()
-        self.user = AuthUserFactory()
-        self.node = ProjectFactory(creator=self.user)
-        self.file1 = api_utils.create_test_file(
-            self.node, self.user, filename='file1')
-        self.payload = {
-            "data": {
-                "type": "files",
-                "id": self.file1._id,
-                "attributes": {
-                    "checkout": None,
-                    "tags": ["goofy"]
+            auth=user.auth,
+        ).status_code == 405
+
+@pytest.mark.django_db
+class TestFileTagging:
+    @pytest.fixture()
+    def node(self, user):
+        return ProjectFactory(creator=user)
+
+    @pytest.fixture()
+    def file_one(self, user, node):
+        return api_utils.create_test_file(
+            node, user, filename='file_one')
+
+    @pytest.fixture()
+    def payload(self, file_one):
+        payload = {
+            'data': {
+                'type': 'files',
+                'id': file_one._id,
+                'attributes': {
+                    'checkout': None,
+                    'tags': ['goofy']
                 }
             }
         }
-        self.url = '/{}files/{}/'.format(API_BASE, self.file1._id)
+        return payload
+
+    @pytest.fixture()
+    def url(self, file_one):
+        return '/{}files/{}/'.format(API_BASE, file_one._id)
 
-    def test_tags_add_properly(self):
-        res = self.app.put_json_api(self.url, self.payload, auth=self.user.auth)
-        assert_equal(res.status_code, 200)
+    def test_tags_add_and_update_properly(self, app, user, url, payload):
```

Comment:
These look like they possibly mutate things. Although the mutations don't overlap it is generally not a great practice to have grouped tests that mutate things. @alexschiller